### PR TITLE
Chore cleanup turbo functions

### PR DESCRIPTION
**Pull request by EvertBunschoten**
## Proposed Changes
Clean up the turbomachinery functions in CEulerSolver.cpp and apply SU2 coding rules.

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [X] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [x] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [x] I used the pre-commit hook to prevent dirty commits and used `pre-commit run --all` to format old commits.
- [x] I have added a test case that demonstrates my contribution, if necessary.
- [x] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

File: SU2_CFD/src/solvers/CEulerSolver.cpp [hunk 1]
```diff
@@ -8577,239 +8577,130 @@ void CEulerSolver::PreprocessAverage(CSolver **solver, CGeometry *geometry, CCon
...
-  for (iSpan= 0; iSpan < nSpanWiseSections + 1; iSpan++){
+  for (auto iSpan= 0; iSpan < nSpanWiseSections + 1; iSpan++){
+    su2double TotalDensity{0}, TotalPressure{0}, TotalNu{0}, TotalOmega{0}, TotalKine{0}, 
+              TotalAreaDensity{0}, TotalAreaPressure{0}, TotalAreaNu{0}, TotalAreaOmega{0}, TotalAreaKine{0},
+              TotalMassDensity{0}, TotalMassPressure{0}, TotalMassNu{0}, TotalMassOmega{0}, TotalMassKine{0};
 
+    su2double *TotalVelocity = new su2double[nDim],
+              *TotalAreaVelocity = new su2double[nDim],
+              *TotalMassVelocity = new su2double[nDim],
+              *TotalFluxes = new su2double[nVar];
```
Comment (pcarruscag) [inline SU2_CFD/src/solvers/CEulerSolver.cpp:8596]: Nice cleanup, you can make these static with size MAXNDIM

File: SU2_CFD/src/solvers/CEulerSolver.cpp [hunk 2]
```diff
@@ -8577,239 +8577,130 @@ void CEulerSolver::PreprocessAverage(CSolver **solver, CGeometry *geometry, CCon
...
-      for (iMarkerTP=1; iMarkerTP < config->GetnMarker_Turbomachinery()+1; iMarkerTP++){
-        if (config->GetMarker_All_Turbomachinery(iMarker) == iMarkerTP){
-          if (config->GetMarker_All_TurbomachineryFlag(iMarker) == marker_flag){
+      auto iPoint = geometry->turbovertex[iMarker][iSpan][iVertex]->GetNode();
 
-            /*--- Retrieve Old Solution ---*/
+      /*--- Retrieve local quantities ---*/
+      auto Pressure = nodes->GetPressure(iPoint);
+      auto Density  = nodes->GetDensity(iPoint);
+      auto Enthalpy = nodes->GetEnthalpy(iPoint);
```
Comment (pcarruscag) [inline SU2_CFD/src/solvers/CEulerSolver.cpp:8614]: Try to add a bit more const if possible.

File: SU2_CFD/src/solvers/CEulerSolver.cpp [hunk 3]
```diff
@@ -8855,155 +8746,149 @@ void CEulerSolver::TurboAverageProcess(CSolver **solver, CGeometry *geometry, CC
...
+            auto TotalArea           = geometry->GetSpanArea(iMarker,iSpan);
+            auto AverageTurboNormal  = geometry->GetAverageTurboNormal(iMarker,iSpan);
+            auto nVert               = geometry->GetnTotVertexSpan(iMarker,iSpan);
 
             /*--- compute normal Mach number as a check for massflow average and mixedout average ---*/
             GetFluidModel()->SetTDState_Prho(TotalAreaPressure/TotalArea, TotalAreaDensity / TotalArea);
-            soundSpeed = GetFluidModel()->GetSoundSpeed();
-            MachTest   = TotalFluxes[0]/(TotalAreaDensity*soundSpeed);
+            auto soundSpeed = GetFluidModel()->GetSoundSpeed();
+            auto MachTest   = TotalFluxes[0]/(TotalAreaDensity*soundSpeed);
```
Comment (pcarruscag) [inline SU2_CFD/src/solvers/CEulerSolver.cpp:8762]: We can't use auto for computations with su2double, it breaks AD

**Comment by pcarruscag** [comment]
@joshkellyjak do you want to take a look too?

**Comment by joshkellyjak** [comment]
Will have a look this week, thanks Evert!

File: SU2_CFD/src/solvers/CEulerSolver.cpp (hunk 3 above)
Comment (EvertBunschoten) [reply SU2_CFD/src/solvers/CEulerSolver.cpp:8750]: Do you mean that  you always should define a variable as `su2double` or `const su2double` when the rhs is a computation?

Comment (pcarruscag) [reply SU2_CFD/src/solvers/CEulerSolver.cpp]: Yep https://su2code.github.io/docs_v7/Style-Guide/

**Review by pcarruscag — APPROVED** [review]

**Pull request merged**